### PR TITLE
planner: add warn log for sync stats

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

add warn log for sync stats and recorded in slowlog

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/column.go [hunk 1]
```diff
@@ -121,9 +121,6 @@ func (c *Column) IsInvalid(sctx sessionctx.Context, collPseudo bool) bool {
 	}
 	if sctx != nil {
 		stmtctx := sctx.GetSessionVars().StmtCtx
-		if stmtctx != nil && stmtctx.StatsLoad.Fallback {
-			return true
-		}
```
Comment (Yisaer) [inline statistics/column.go:126]: We don't need to check `Fallback` here anymore, we can use `StatsLoadedStatus` instead

File: statistics/handle/handle_hist.go [hunk 2]
```diff
@@ -104,6 +119,8 @@ func (h *Handle) SyncWaitStatsLoad(sc *stmtctx.StatementContext) bool {
 			}
 		case <-timer.C:
 			metrics.SyncLoadTimeoutCounter.Inc()
+			_, digest := sc.SQLDigest()
+			logutil.BgLogger().Warn("SyncWaitStatsLoad timeout", zap.String("digest", digest.String()))
```
Comment (hawkingrei) [inline statistics/handle/handle_hist.go:123]: We can print the timeout value here.

Comment (Yisaer) [reply statistics/handle/handle_hist.go:123]: This is not necessary as we stored timeout duration in session vars already.

File: planner/core/plan_stats.go [hunk 3]
```diff
@@ -60,6 +64,7 @@ func (s syncWaitStatsLoadPoint) optimize(_ context.Context, plan LogicalPlan, _
 	if plan.SCtx().GetSessionVars().InRestrictedSQL {
 		return plan, nil
 	}
+	// For now, SyncWaitStatsLoad won't return error due to handleTimeout will handle the timeout error and transfer into warning
```
Comment (chrysan) [inline planner/core/plan_stats.go:67]: ditto.

File: statistics/handle/handle_hist.go [hunk 4]
```diff
@@ -46,14 +46,14 @@ type StatsLoad struct {
...
-	WorkingColMap  map[model.TableItemID][]chan model.TableItemID
+	WorkingColMap  map[model.TableItemID][]chan stmtctx.StatsLoadResult
 }
 
 // NeededItemTask represents one needed column/indices with expire time.
 type NeededItemTask struct {
-	TableItemID model.TableItemID
-	ToTimeout   time.Time
-	ResultCh    chan model.TableItemID
+	TableItemResult stmtctx.StatsLoadResult
```
Comment (chrysan) [inline statistics/handle/handle_hist.go:54]: It's a little strange that task contains result, is TableItemID enough? And wrap it to be TableItemResult when write response.

File: planner/core/plan_stats.go [hunk 5]
```diff
@@ -85,6 +90,9 @@ func RequestLoadStats(ctx sessionctx.Context, neededHistItems []model.TableItemI
 	var timeout = time.Duration(waitTime)
 	err := domain.GetDomain(ctx).StatsHandle().SendLoadRequests(stmtCtx, neededHistItems, timeout)
 	if err != nil {
+		_, digest := stmtCtx.SQLDigest()
+		logutil.BgLogger().Warn("SendLoadRequests failed",
+			zap.String("digest", digest.String()), zap.Error(err))
```
Comment (chrysan) [inline planner/core/plan_stats.go:95]: What about write this warn message to slow logs or statement summary?

**Review by chrysan — CHANGES_REQUESTED** [review]

File: planner/core/plan_stats.go (hunk 5 above)
Comment (Yisaer) [reply planner/core/plan_stats.go:95]: I don't think record error detail in slowlog is a good choice, but I think it's proper to record a flag if sync stats failed in slow log.

**Review by chrysan — APPROVED** [review]
Please test the failure path by UT or FT. Rest LGTM.

**Comment by Yisaer** [comment]
> Please test the failure path by UT or FT. Rest LGTM.

updated

File: sessionctx/stmtctx/stmtctx.go [hunk 6]
```diff
@@ -1014,3 +1018,32 @@ func (d *CopTasksDetails) ToZapFields() (fields []zap.Field) {
...
+	if r.Error == nil {
+		return ""
+	}
+	b := bytes.NewBufferString("tableID:")
+	b.WriteString(strconv.FormatInt(r.Item.TableID, 10))
+	b.WriteString(", id:")
+	b.WriteString(strconv.FormatInt(r.Item.ID, 10))
+	b.WriteString(", isIndex:")
+	b.WriteString(strconv.FormatBool(r.Item.IsIndex))
+	if r.Error != nil {
```
Comment (xuyifangreeneyes) [inline sessionctx/stmtctx/stmtctx.go:1044]: We already check `r.Error == nil` at line 1035.

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**